### ISSUE TITLE
Monorepo support with typedoc config "entryPointStrategy": "packages"

### ISSUE THREAD

**Issue by joshuaboud** [krisztianb/typedoc-umlclass#35]
Hi @krisztianb  
  
I can get the plugin to generate images just when ran for a single package, but images do not seem to be generated with a monorepo setup. Here is the repo I am working with for example: https://github.com/45Drives/houston-common.  
  
Running `yarn docs` from the monorepo root yields:
```
[info] Loaded plugin typedoc-umlclass
[info] Converting project at ./houston-common-lib
2024-04-29, 4:38:49 p.m. [INFO] Caculating number of diagrams to generate ...
2024-04-29, 4:38:49 p.m. [INFO] The result is: 22
2024-04-29, 4:38:49 p.m. [INFO] Setting up progress bar ...
[info] Converting project at ./houston-common-ui
[warning] No entry points were provided, this is likely a misconfiguration.
2024-04-29, 4:38:49 p.m. [INFO] Caculating number of diagrams to generate ...
2024-04-29, 4:38:49 p.m. [INFO] The result is: 0
[info] Merging converted projects
[info] Documentation generated at ./docs
```
(nothing happening after "Setting up progress bar ...")  
  
However `yarn docs` run from `./houston-common-lib` yields:
```
[info] Loaded plugin typedoc-umlclass
2024-04-29, 4:40:21 p.m. [INFO] Caculating number of diagrams to generate ...
2024-04-29, 4:40:21 p.m. [INFO] The result is: 22
2024-04-29, 4:40:21 p.m. [INFO] Setting up progress bar ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection BasicError ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection BasicError ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection BasicError ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection BasicError ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Channel ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Channel ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Channel ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Channel ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection ChannelEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection ChannelEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection ChannelEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection ChannelEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection CockpitEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection CockpitEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection CockpitEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection CockpitEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection ControlMessage ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection ControlMessage ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection ControlMessage ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection ControlMessage ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection DBusProxy ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection DBusProxy ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection DBusProxy ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection DBusProxy ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection DBusProxyEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection DBusProxyEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection DBusProxyEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection DBusProxyEvents ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection DeferredPromise ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection DeferredPromise ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection DeferredPromise ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection DeferredPromise ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection EventMap ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection EventMap ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection EventMap ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection EventMap ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection EventSource ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection EventSource ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection EventSource ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection EventSource ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Spawn ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Spawn ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Spawn ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Spawn ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection SpawnException ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection SpawnException ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection SpawnException ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection SpawnException ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection BashCommand ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection BashCommand ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection BashCommand ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection BashCommand ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Command ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Command ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Command ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Command ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Directory ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Directory ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Directory ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Directory ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection ExitedProcess ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection ExitedProcess ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection ExitedProcess ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection ExitedProcess ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection File ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection File ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection File ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection File ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection NonZeroExit ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection NonZeroExit ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection NonZeroExit ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection NonZeroExit ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Path ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Path ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Path ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Path ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection Process ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection Process ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection Process ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection Process ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection ProcessBase ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection ProcessBase ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection ProcessBase ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection ProcessBase ...
2024-04-29, 4:40:21 p.m. [INFO] Processing page for reflection ProcessError ...
2024-04-29, 4:40:21 p.m. [INFO] Generating PlantUML code for reflection ProcessError ...
2024-04-29, 4:40:21 p.m. [INFO] Creating leged for diagram of reflection ProcessError ...
2024-04-29, 4:40:21 p.m. [INFO] Creating diagram image for reflection ProcessError ...
2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection CockpitEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection CockpitEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection CockpitEvents ...
Using 64 process(es) to add 22 class diagrams [==--------------------------------------] 4%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection DBusProxyEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection DBusProxyEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection DBusProxyEvents ...
Using 64 process(es) to add 22 class diagrams [====------------------------------------] 9%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection ChannelEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection ChannelEvents ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection ChannelEvents ...
Using 64 process(es) to add 22 class diagrams [=====-----------------------------------] 13%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection ControlMessage ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection ControlMessage ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection ControlMessage ...
2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection Directory ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection Directory ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection Directory ...
Using 64 process(es) to add 22 class diagrams [=========-------------------------------] 22%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection BasicError ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection BasicError ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection BasicError ...
Using 64 process(es) to add 22 class diagrams [===========-----------------------------] 27%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection SpawnException ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection SpawnException ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection SpawnException ...
2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection ExitedProcess ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection ExitedProcess ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection ExitedProcess ...
Using 64 process(es) to add 22 class diagrams [===============-------------------------] 36%2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection Command ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection Command ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection Command ...
2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection Path ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection Path ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection Path ...
2024-04-29, 4:40:22 p.m. [INFO] Writing local image file for reflection File ...
2024-04-29, 4:40:22 p.m. [INFO] Creating local image URL for reflection File ...
2024-04-29, 4:40:22 p.m. [INFO] Inserting diagram into page for reflection File ...
Using 64 process(es) to add 22 class diagrams [====================--------------------] 50%2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection Channel ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection Channel ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection Channel ...
Using 64 process(es) to add 22 class diagrams [======================------------------] 54%2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection DBusProxy ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection DBusProxy ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection DBusProxy ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection NonZeroExit ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection NonZeroExit ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection NonZeroExit ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection EventMap ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection EventMap ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection EventMap ...
Using 64 process(es) to add 22 class diagrams [===========================-------------] 68%2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection Process ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection Process ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection Process ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection BashCommand ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection BashCommand ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection BashCommand ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection ProcessError ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection ProcessError ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection ProcessError ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection ProcessBase ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection ProcessBase ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection ProcessBase ...
Using 64 process(es) to add 22 class diagrams [===================================-----] 86%2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection Spawn ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection Spawn ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection Spawn ...
2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection EventSource ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection EventSource ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection EventSource ...
Using 64 process(es) to add 22 class diagrams [======================================--] 95%2024-04-29, 4:40:23 p.m. [INFO] Writing local image file for reflection DeferredPromise ...
2024-04-29, 4:40:23 p.m. [INFO] Creating local image URL for reflection DeferredPromise ...
2024-04-29, 4:40:23 p.m. [INFO] Inserting diagram into page for reflection DeferredPromise ...
Using 64 process(es) to add 22 class diagrams [========================================] 100%
2024-04-29, 4:40:23 p.m. [INFO] Attaching content to CSS file ...
2024-04-29, 4:40:23 p.m. [INFO] DONE
[info] Documentation generated at /home/jboudreau/git/cockpit-file-sharing/houston-common/docs
```

**Comment by krisztianb** [comment]
Hi. I've not used this plugin for monorepos yet. But I've had problems with my other TypeDoc plugins in the past when using monorepos because the behavior (events etc.) of TypeDoc is a bit different in that case. This is probably also causing issues here. I will have to investigate this. Thank you for providing an example repository.

**Comment by krisztianb** [comment]
Now that I looked at it I think this is actually a configuration problem of TypeDoc within your repository.

I created a test setup in a branch here: https://github.com/krisztianb/typedoc-umlclass/tree/GH-35-failure-in-monorepo/test/monorepo-test

It is based on https://github.com/Gerrit0/typedoc-packages-example as described here (see point "packages"): https://typedoc.org/options/input/#entrypointstrategy

Within this test branch I can run `npx typedoc` in the main folder `monorepo-test` and the subfolders `monorepo-test/input/project-1` and `monorepo-test/input/project-2` getting the expected documentation which also contains the class diagrams.

Could you check your TypeDoc configuration?

**Comment by joshuaboud** [comment]
I looked into it and I think I figured it out. My root typedoc.json entryPoints was `["houston-common-*"]`, but I had not set up typedoc within houston-common-ui yet. Changing entryPoints to `["houston-common-lib"]` now properly generates the images.  
My issue is fixed anyway so feel free to close, but it still might be worth looking into why output for package A is not generated when package B is misconfigured. Thank you for the help!
